src/utils/hooks.ts

```diff
@@ -0,0 +1,104 @@
+'use client';
+
+import { useCallback, useContext, useEffect, useMemo } from 'react';
+import { usePathname, useRouter } from 'next/navigation';
+import { OnboardingContext } from './OnboardingProvider';
+
+/**
+ * Wrapper to use the onboarding context.
+ * Throws an error if onboarding context doesn't exist
+ * (i.e. the component is not wrapped in a onboarding provider, or it's a server component)
+ * @returns the onboarding context object.
+ */
+const useSafeOnboarding = () => {
```

Comment:
naming might not be clear, as this is safeonboarding, but the next is guarded. but they serve different purposes, one on context and one on the actual onbaording